accord-core/src/main/java/accord/local/CommandStore.java

```diff
@@ -83,6 +123,65 @@ public Agent agent()
 
     public abstract void shutdown();
 
+    // implementations are expected to override this for persistence
+    protected void setRejectBefore(ReducingRangeMap<Timestamp> newRejectBefore)
+    {
```

Comment:
could you make this abstract, since the implementor is expected to override?

Reply:
same for `setBootstrapBeganAt` and `setSafeToRead`

Reply:
I can introduce an additional method, perhaps, though we probably need to think this through when we integrate anyway so it probably doesn't help much yet. But, to explain my thinking here: all of the implementation details are kept within Accord so that the implementation cannot mess with it (though of course it must invoke `super`). I don't really want the implementation controlling the data structure itself, just to have the ability to persist.